### ISSUE TITLE
Purifier you craft is now "human flesh"

### ISSUE THREAD

**Issue by toweroglass** [CleverRaven/Cataclysm-DDA#56937]
### Describe the bug

Just updated my game. was current as of less than a week ago. now when i craft purifier or injectable purifier the game tells me i'm going to consume human flesh. then it gives me the debuff.
made a test character and got similar results.

### Steps to reproduce

1. make new character.
2. debug in purifier. see it is not made of people.
3. debug everything you need to make purifier and craft purifier. it is made of people. see screenshot below.

that last step was a pain to reproduce.

### Expected behavior

as of a week ago my character came to terms with the fact that humans were sacrificed for them to make purifier. not to mention many of these samples came from a TCL storage. 

Getting the debuff when i hack up a feral sounds fair, but once i make the purifier i don't really know what sample came from where.

### Screenshots

![mutagen](https://user-images.githubusercontent.com/81603517/164125713-91007aad-acb2-4bd2-a2ac-f906b2956c7b.PNG)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1645 (21H2)
- Game Version: 3ad05f9 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Alternative Map Key [alt_map_key],
    sees-player, run, crouch, bleed icons, retrodays [sees_player_retro],
    No Fungal Growth [no_fungal_growth]
]

**Comment by Qrox** [comment]
If that human sample came from someone you killed I'd say it's reasonable for it to have a morale debuff. Maybe we can add an interaction with friendly non-mutated NPC that asks them to provide you with a sample if they trust you enough, which will then not contain the cannibalism flag.

**Comment by OldFriendEr** [comment]
isn't a purifier a human stem cell treatment? I wonder how a NPC would help with that

**Comment by Zireael07** [comment]
@OldFriendEr Donating tissue/cell samples

**Comment by toweroglass** [comment]
> If that human sample came from someone you killed I'd say it's reasonable for it to have a morale debuff. Maybe we can add an interaction with friendly non-mutated NPC that asks them to provide you with a sample if they trust you enough, which will then not contain the cannibalism flag.

wouldn't the game need to track whether I killed it or not? I can also get samples from the TCL storage that I am far removed from the source. 
in the debug world i made purifier i crafted from human samples was human flesh. purifier i spawned in was not human flesh. in neither case did i butcher a single soul.

why would my character think they are eating human flesh? they are injecting it and they've already dealt with their misgivings having been upset about butchering the feral.

**Comment by fairyarmadillo** [comment]
I don't know, injecting yourself with stem cells stolen from a human that you murdered (self defense or no) does seem tangibly different from ethically-sourced donor cells. I'd almost consider it worse than cannibalism (though it's less visceral) because most of the time, mutation is a vanity/self-improvement project rather than something necessary for your immediate survival.

The scientists involved in developing mutagen were committing Rimworld-tier crimes against humanity, following in their footsteps should be upsetting unless you're a psychopath or too mutated to care.

**Comment by toweroglass** [comment]
> I don't know, injecting yourself with stem cells stolen from a human that you murdered (self defense or no) does seem tangibly different from ethically-sourced donor cells. I'd almost consider it worse than cannibalism even though it's less visceral because most of the time, mutation is a vanity/self-improvement project rather than something necessary for your immediate survival.

Like I tried to hash out. the acknowledgement of the atrocity should be covered fairly well by the need to get over the fact i had to butcher another person. constantly pinging my morale over the same issue seems excessive. 

> 
> The scientists involved in developing mutagen were committing Rimworld-tier crimes against humanity, 

We have a real world analogy for this because we all benefit from extremely unethical practices from the past and we don't bat an eye at it. We don't see the atrocities and we reap the rewards. which even if I agree that my character should be constantly beating himself up over it, it doesn't have to be material I disected from another human. it does the same for the freebies i found in storage which are not my concern.

> following in their footsteps should be upsetting unless you're a psychopath or too mutated to care.

if one has significant negative mutations you would think that would be enough for them to desperately try something to revert them without getting too sad. also, purifier i spawn in doesn't give me the ate a human now i'm sad debuff. shouldn't it be the same thing with this justification?


anyways, I don't really care to be a part of the conversation if this is an intentional change. I can't find the PR that addresses it so I thought it was a bug. If this conversation has been had then I won't bring it up again.

**Comment by fairyarmadillo** [comment]
We have ethically-sourced stuff and stuff that came from murder, obviously the debug menu is spawbing the former. Debug-spawned purifier also isn't a part of the simulation and doesn't really factor here.

I take medicine in real life which I'm sure has its roots in some unethical practice at some point in it's R&D, but it doesn't contain human flesh collected from murder victims, especially not ones I or one of my friends killed and butchered, so that's not really a valid analogy.

Lastly the bit about being desperate enough to ignore feeling guilty would also apply to survival-driven cannibalism too, wouldn't it? But we know from real life accounts involving people resorting to cannibalism that it doesn't work that way. Survival might motivate a typical person to get over their reluctance to consume human flesh, but they still tend to feel bad about it after.

**Comment by toweroglass** [comment]
> We have ethically-sourced samples (stuff you find in labs, which would have been easy for a major biocorp company to get pre-cataclysm) and samples that came from murder, obviously the debug menu is spawning the former. Debug-spawned purifier also isn't a part of the simulation and doesn't really factor here. You can use the debug menu to do all kinds of things that don't make sense.
> 
> I take medicine in real life which I'm sure has its roots in some unethical practice at some point in it's R&D, but it doesn't contain human flesh collected from murder victims, especially not ones I or one of my friends killed and butchered, so that's not really a valid analogy.

You yourself made the point that the human samples from the lab are unethically sourced. I was responding to your point in both instances.

**Comment by fairyarmadillo** [comment]
> You yourself made the point that the human samples from the lab are unethically sourced. I was responding to your point in both instances.

I did not say that, you were just cherry picking to try to support your point.

The labs wod have had access to sources of human tissue that did not require anyone to die. These would be standard tissue and stem cell (not fetal stem cell) samples that any real life lab could get right now. The player does not (currently) have access to new sources of this kind of ethically-sourced material, and has to find them in labs. Sourcing these samples instead from the murder of humans drastically changes the ethical question of the purifier itself.

In other words: purifier made from willingly-given donor tissue that didn't hurt the donor is very different from purifier made out of a guy you beat to death with a hammer. If the player makes one of the latter and gives it to an NPC, that NPC should feel upset about what they're drinking/injecting.

You should be able to lie about the source (same goes for dishes containing human meat or other things someone might not want to eat), but that's a separate issue.

**Comment by toweroglass** [comment]
I was responding to "they were committing Rimworld-tier crimes against humanity". As in real life, the harsh realities of scientific developments are out of site out of mind so the very least human samples procured in the wild should be viewed differently than one you butchered someone for. 

I thought i was spawning it in the same as materials found at a lab but maybe I'm wrong. I'm not up to creating a new world and finding a lab. 

i really don't have a horse in the race. It looked like a bug to me because i performed the same action a week ago and got different results. or maybe I just _think_ i did. that's probably my bad. whatever the dev team decided is canon is whatever and i'm not arguing with it.

**Comment by fairyarmadillo** [comment]
I went through and tried making purifier from debug, spawned, and harvested human samples, and the resulting purifier all counts as cannibalism, so that was my mistake. IMO the lab-spawned samples should be guilt-free, but that's maybe a discussion for another issue.

[There's some discussion here,](https://github.com/CleverRaven/Cataclysm-DDA/pull/53279) but otherwise the issue's seemingly up for debate. Logical consistency aside, I'd personally prefer if it _wasn't_ cannibalism because it's a pain in the ass to deal with in game, but it doesn't seem like a bug to me.

**Comment by toweroglass** [comment]
yeah sorry if i seemed irritable. I'm not going to argue with the direction of the devs. just didn't feel understood.

I scanned the thread for relevant purifier discussion. I didn't see anyone explicitly agreeing that the serum itself should amount to cannibalism, cause to be clear regardless i think hacking up a human for the sample triggering a debuff is perfectly reasonable.

Ultimately if purifier, homemade or otherwise, equals cannibalism was the intent I'm happy to close this issue.
